include/data/json.h

```diff
@@ -173,6 +173,54 @@ typedef int (*json_append_bytes_t)(const char *bytes, size_t len,
 		}, \
 	}
 
+/**
+ * @brief Internal helper macro to declare an element descriptor
+ *
+ * @param struct_ Struct packing the values
+ * @param len_field_ Field name in the struct for the number of elements
+ * in the array
+ * @param elem_type_ Element type, must be a primitive type
+ * @param union_ Optional macro argument containing array or object descriptor
+ */
+#define Z_JSON_ELEMENT_DESCR(struct_, len_field_, elem_type_, union_) \
+	(const struct json_obj_descr[1]) \
```

Comment:
Whether `const struct json_obj_descr[]` or `const struct json_obj_descr[1]` is better/valid is beyond my C vs. C++ knowledge. Maybe @pabigot could have a quick look, as you provided related PRs for Bluetooth?

Reply:
Since when I fixed the C++ problem with using an address of temporary I used `type[]` I don't know of any reason why it should change to `type[1]`, but newer versions of GNUC might have a problem.  In context `[]` should just cause the compiler to infer the array length from the single-element value.

Reply:
Agreed, there really shouldn't be a difference between ```type[]``` and ```type[1]``` as compiler will infer the length. Specifying the length could cause maintenance issue down the road, I'll update to ```const struct json_obj_descr[]```. I really  appreciate the insight